### PR TITLE
Add ClashX Pro.app v1.40.0.1

### PR DESCRIPTION
**Pull request by gqbre**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).
- [x] `brew audit --cask {{cask_file}}` is error-free.
- [x] `brew style --fix {{cask_file}}` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask {{cask_file}}` worked successfully.
- [x] `brew install --cask {{cask_file}}` worked successfully.
- [x] `brew uninstall --cask {{cask_file}}` worked successfully.

### CODE REVIEW

File: Casks/clashx-pro.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+cask "clashx-pro" do
+  version "1.40.0.1"
+  sha256 "4f9fb11bf8118478e1f3e52c951cc3f6a6e618c043377f6e0e543656344f59dd"
+
+  url "https://appcenter.vercel.app/clashx/clashx-pro/#{version}",
+      verified: "appcenter.vercel.app/"
+  appcast "https://api.appcenter.ms/v0.1/public/sparkle/apps/1cd052f7-e118-4d13-87fb-35176f9702c1"
```
Comment (vitorgalvao) [inline Casks/clashx-pro.rb:7]: Are you able to try a [`livecheck`](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/livecheck.md) instead?

File: Casks/clashx-pro.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  name "ClashX Pro"
+  desc "Rule-based custom proxy with GUI based on clash [Pro Version]"
```
Comment (vitorgalvao) [inline Casks/clashx-pro.rb:9]:
```suggestion
  desc "Rule-based custom proxy with GUI based on clash"
```

**Comment by vitorgalvao** [comment]
Welcome! Great first contribution. Please do let us know if you have difficulty with the request.

File: Casks/clashx-pro.rb (hunk 1 above)
Comment (gqbre) [reply Casks/clashx-pro.rb]: Done

**Comment by gqbre** [comment]
All done，check please @vitorgalvao

File: Casks/clashx-pro.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+cask "clashx-pro" do
+  version "1.40.0.1"
+  sha256 "4f9fb11bf8118478e1f3e52c951cc3f6a6e618c043377f6e0e543656344f59dd"
+
+  url "https://appcenter.vercel.app/clashx/clashx-pro/#{version}",
```
Comment (vitorgalvao) [inline Casks/clashx-pro.rb:5]: How are you arriving at this specific URL?

Comment (gqbre) [reply Casks/clashx-pro.rb:5]: It created by https://github.com/rx-ts/appcenter
Create stable link that redirects to unstable App Center download link via Vercel with puppeteer.
Thanks for @JounQin

Comment (JounQin) [reply Casks/clashx-pro.rb:5]: According to https://github.com/Homebrew/homebrew-cask/pull/79549#issuecomment-725521358, we should not use this unofficial service.

And here is my another approach https://github.com/Homebrew/homebrew-cask/pull/79549#issuecomment-735386385.

See https://github.com/homebrew-x/homebrew-x/blob/master/Casks/clashx-pro.rb

Comment (vitorgalvao) [reply Casks/clashx-pro.rb:5]: Yep, we can only take official URLs in this repo. Everything else is a possible security risk and we want casks to be auditable by regular users. That’s why the `verified` parameter exists.

**Pull request closed without merging**